src/setup.cpp

```diff
@@ -449,6 +449,8 @@ void config_menu(void)
  */
 static bool getakey(KPlayerInput::button& b, const char* cfg)
 {
+    // Wait for all keys to be released
+    Game.wait_released();
```

Comment:
There needs to be another `Game.wait_released()` right before the `return true` or else it "prompts" the user again if the user doesn't lift their finger fast enough.

And as a UX question/discussion more than a nit: would you want to move this to just after the "Press a key" prompt appears, instead of before? The only real difference is that you would see that prompt immediately while it waits for the user to release the initial key press, instead of only seeing the prompt after.

Reply:
```diff
diff --git a/src/setup.cpp b/src/setup.cpp
index f858988..2e8bb0e 100644
--- a/src/setup.cpp
+++ b/src/setup.cpp
@@ -448,10 +448,11 @@ void config_menu(void)
 static bool getakey(KPlayerInput::button& b, const char* cfg)
 {
     // Wait for all keys to be released
-    Game.wait_released();
     Draw.menubox(double_buffer, 108, 108, 11, 1, DARKBLUE);
     Draw.print_font(double_buffer, 116, 116, _("Press a key"), FNORMAL);
     Draw.blit2screen();
+    Game.wait_released();
+
     // Wait 5 secs then give up
     int timeout = Game.KQ_TICKS * 5;
     while (timeout > 0)
@@ -469,6 +470,8 @@ static bool getakey(KPlayerInput::button& b, const char* cfg)
                     b.scancode = a;
                     Config.set_config_int(NULL, cfg, a);
                 }
+                // Wait to close the "Press a key" dialog until the user releases the new key.
+                Game.wait_released();
                 return true;
             }
         }
```

Reply:
I tried both and slightly preferred showing the prompt after the keys were released, but it didn't really matter.
Updated the PR to make those changes.